### PR TITLE
use react-timers throughout

### PR DESCRIPTION
**Pull request by dcousens**
This fixes a tonne of errors where these timeouts were being called when the component was no longer mounted.

To reproduce, just watch the console while moving around in the examples.

### CODE REVIEW

File: src/js/views/component/bar-header-search.js [hunk 1]
```diff
@@ -2,38 +2,40 @@ var React = require('react'),
 	SetClass = require('classnames'),
 	Navigation = require('touchstonejs').Navigation,
 	Tappable = require('react-tappable'),
-	Link = require('touchstonejs').Link,
```
Comment (dcousens) [inline src/js/views/component/bar-header-search.js]: Unused variable

**Comment by JedWatson** [comment]
nice work :+1:

File: src/js/views/details.js [hunk 2]
```diff
@@ -1,14 +1,13 @@
 var React = require('react'),
-	Link = require('touchstonejs').Link,
 	Tappable = require('react-tappable'),
 	Dialogs = require('touchstonejs').Dialogs,
 	Navigation = require('touchstonejs').Navigation,
 	UI = require('touchstonejs').UI;
 
-var moment = require('moment')
```
Comment (dcousens) [inline src/js/views/details.js]: Again, unused variables, probably should be in a separate commit
